File: hw/palomo/u3/hw26Observer/Observer/src/main/java/ec/edu/espe/observer/view/StockView.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package ec.edu.espe.observer.view;

import ec.edu.espe.observer.model.Stock;
/**
 *
 * @author Cristian Palomo, Error 404, @ESPE
 */
public class StockView {
    public static void printNotification(String investorName, Stock stock, Object args) {
        System.out.println("Notified observer " + investorName);
        if (args instanceof String) {
            System.out.println("The symbol of " + stock.getSymbol() + " changed to: " + args);
        } else if (args instanceof Double) {
            System.out.println("The price of " + stock.getSymbol() + " changed to: " + args);
        }
    }
}

File: hw/basurto/u3/hw26Observer/hw26Observer.py/view/stock_view.py
class StockView:
    @staticmethod
    def print_notification(investor_name, stock, args):
        print(f"Notified observer {investor_name}")
        if isinstance(args, str):
            print(f"The symbol of {stock.symbol} changed to: {args}")
        elif isinstance(args, (int, float)):
            print(f"The price of {stock.symbol} changed to: {args:.2f}")